### PR TITLE
[sca, kmac] Add new capture script for KMAC

### PR DESCRIPTION
**Pull request by m-temp**
This PR addresses https://github.com/lowRISC/ot-sca/issues/217

As this was a major rework, it would be good if s.o. could test if the TVLA still works.

There are three open points:
 1.  FIXMEs in the config file: I'm not sure how to translate the cycles vs. samples
 2. In the non-bach fvsr-mode, the internal PRNG is used not AES. This is the same behavior as in the old-script. However we should adjust this eventually.
 3. The keyword "ciphertext" is used throughout the script. We should change this to mac or tag eventually.

### CODE REVIEW

**Comment by nasahlpa** [comment]
Thanks @m-temp, I will have a closer look tomorrow.

Regarding the samples vs. cycles. You either can specify whether you want to configure the Husky scope in samples or cycles. If both are provided, the samples are taken.
In the AES script, we configured the scope in cycles, the [cycle converter](https://github.com/lowRISC/ot-sca/blob/master/capture/scopes/cycle_converter.py) then converts the cycles into samples (which is needed by Husky).

So if you know how many cycles a KMAC run takes, you probably want to configure the num_cycles parameter and remove the num_samples.

**Comment by nasahlpa** [comment]
Thanks for changing the configuration file! Capture trace looks IMHO good. Could you please try to run TVLA?

**Comment by vrozic** [comment]
Thanks @m-temp . Great work

I've tested all three capture types locally with my setup and they all work.

TVLA works well with one caveat. At the moment TVLA figures of general test results contain meta-data that are read from the project file. This reports an error when we use the new data-base format. After slightly modifying the `tvla.py` I got this figure:
kmac_fixed_vs_random

The results look as expected.
Since the TVLA problem is unrelated, I will solve it in a separate PR. I am tracking this in issue #72

**Review by vrozic — APPROVED** [review]
In my view this is ready to merge, modulo two small comments.
We should open an issue to remember to add this capture to CI later on.

File: capture/lib/ot_communication.py [hunk 1]
```diff
@@ -93,3 +93,60 @@ def read_ciphertext(self, len_bytes):
...
+            key: Bytearray containing the key.
+        """
+        self.target.simpleserial_write("k", key)
+
+    def fvsr_key_set(self, key):
+        """ Write key to KMAC.
+        Args:
+            key: Bytearray containing the key.
+        """
+        self.target.simpleserial_write("f", key)
```
[the remark this answers is not in the document]
Comment (m-temp) [reply capture/lib/ot_communication.py]: I've updated the comment for the second one to "Write the fixed key to KMAC".
However, I just noticed, that for AES the description (and the functionality?) is the same.

**Pull request merged**